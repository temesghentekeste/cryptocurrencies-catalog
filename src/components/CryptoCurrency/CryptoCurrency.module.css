.cryptoCurrencyCard {
  width: 15rem;
  margin-bottom: 1.4rem;
  margin-right: 2rem;
  padding: 1rem;
  border: 1px solid gray;
  background: #444;
  color: white;
  border-radius: 5px;
}

.cryptoCurrencyCard__symbol {
  margin-bottom: 1rem;
  padding: 0.4rem 0.6rem;
  background: #998695;
  color: #fff;
  font-weight: 700;
}

.cryptoCurrencyCard__body {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.cryptoCurrencyCard button {
  width: 100%;
  padding: 0.6rem 0.8rem;
  margin: 1rem 0 0;
  background: rgb(75, 95, 73);
  border: none;
  border-radius: 5px;
  font-weight: 700;
  color: #fff;
}

.cryptoCurrencyCard img {
  display: block;
  width: 90%;
  margin: 1rem auto;
}

@media screen and (max-width: 982px) {
  .cryptoCurrencyCard {
    width: 20rem;
  }
}

@media screen and (max-width: 850px) {
  .cryptoCurrencyCard {
    width: 18rem;
  }
}

@media screen and (max-width: 768px) {
  .cryptoCurrencyCard {
    width: 100%;
  }
}
