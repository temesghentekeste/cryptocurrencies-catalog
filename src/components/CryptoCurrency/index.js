/* eslint-disable no-unused-vars */
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './CryptoCurrency.module.css';
import mainStyle from '../../index.module.css';

const CryptoCurrency = ({
  symbol,
  name,
  currency,
  stockExchange,
  exchangeShortName,
}) => (
  <Link to={`/cryptoquote/${symbol}`}>
    <div className={styles.cryptoCurrencyCard}>
      <header>
        <p className={mainStyle.badge}>{name}</p>
        <p className={styles.cryptoCurrencyCard__symbol}>{symbol}</p>
      </header>

      <div className={styles.cryptoCurrencyCard__body}>
        <span>{currency}</span>
        <span>{stockExchange}</span>
        <span>{exchangeShortName}</span>
      </div>

      <footer>
        <button type="button">More...</button>
      </footer>
    </div>
  </Link>
);

CryptoCurrency.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  stockExchange: PropTypes.string.isRequired,
  exchangeShortName: PropTypes.string.isRequired,
};
export default CryptoCurrency;
