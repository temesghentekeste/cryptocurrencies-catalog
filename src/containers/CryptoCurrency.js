import React from 'react';

const CryptoCurrency = ({
  symbol,
  name,
  currency,
  stockExchange,
  exchangeShortName,
}) => (
  <div className="cryptoCurrency">
    <header>
      <p>{name}</p>
      <p className="cryptoCurrency__symbol">{symbol}</p>
    </header>

    <footer>
      <span>{currency}</span>
      <span>{stockExchange}</span>
      <span>{exchangeShortName}</span>
    </footer>
  </div>
);

export default CryptoCurrency;
