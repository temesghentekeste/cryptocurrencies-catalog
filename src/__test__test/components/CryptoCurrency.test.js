import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CryptoCurrency from '../../components/CryptoCurrency';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Router>
      <CryptoCurrency
        id={'BitCoin'}
        symbol="BitCoin"
        name="BitCoin"
        price={1.25}
        image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
      />
    </Router>
  );
  getByTestId = component.getByTestId;
});

it('renders the CryptoCurrency component', () => {
  const component = renderer
    .create(
      <Router>
        <CryptoCurrency
          id={'BitCoin'}
          symbol="BitCoin"
          name="BitCoin"
          price={1.25}
          image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
        />
      </Router>
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});
