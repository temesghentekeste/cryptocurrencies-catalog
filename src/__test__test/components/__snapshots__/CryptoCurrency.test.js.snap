// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CryptoCurrency component 1`] = `
<a
  href="/cryptoquote/BitCoin"
  onClick={[Function]}
>
  <div
    className="cryptoCurrencyCard"
    data-testid="cryptoCurrency"
  >
    <header>
      <p
        className="badge"
      >
        BitCoin
      </p>
      <p
        className="cryptoCurrencyCard__symbol"
      >
        BitCoin
      </p>
    </header>
    <div
      className="cryptoCurrencyCard__body"
    >
      <span>
        Price: $
        1.25
      </span>
      <img
        alt="BitCoin"
        src="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
      />
    </div>
    <footer>
      <button
        type="button"
      >
        More...
      </button>
    </footer>
  </div>
</a>
`;

exports[`renders the Header component 1`] = `
<header
  className="header"
  data-testid="header"
>
  <div
    className="header__right"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Cryptocurrencies
      </h1>
    </a>
    <a
      href="/trending"
      onClick={[Function]}
    >
      <h3>
        Trending Dashboard
      </h3>
    </a>
  </div>
  <div
    className="header__search"
  >
    <input
      name="keyword"
      onChange={[Function]}
      placeholder="Search for something. Ex. iExec, Bitcoin, DOGEUSD..."
      type="text"
      value=""
    />
    <form
      onSubmit={[Function]}
    >
      <input
        className="btnSubmit"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</header>
`;
